cpp/src/lance/format/schema.cc

```diff
@@ -172,6 +173,13 @@ ::arrow::Status Field::LoadDictionary(std::shared_ptr<::arrow::io::RandomAccessF
   return set_dictionary(dict_arr);
 }
 
+int32_t Field::GetFieldsCount() const {
```

Comment:
So this is a flattened count right?

Reply:
This is getting the node counts of the entire schema tree.